src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -548,13 +553,14 @@ private static Long findSchema(Connection connection, BinlogPosition targetPosit
 			PreparedStatement s = connection.prepareStatement(
 				"SELECT id from `schemas` "
 				+ "WHERE deleted = 0 "
-				+ "AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position <= ?)) AND server_id = ? "
-				+ "ORDER BY binlog_file DESC, binlog_position DESC limit 1");
+				+ "AND last_heartbeat_read <= ? AND ((binlog_file < ?) OR (binlog_file = ? and binlog_position <= ?)) AND server_id = ? "
+				+ "ORDER BY last_heartbeat_read DESC, binlog_file DESC, binlog_position DESC limit 1");
```

Comment:
It only really comes into it if you have multiple schemas saved within a single heartbeat interval (typically a second or two). In that case I think the binlog position should be the best ordering - old recycled binlogs should be well in the past, but it could be possible for two maxwell clients to create schema snapshots at roughly the same time (in the wrong order), if one is slightly lagged. Both cases do seem very unlikely though 🤷‍♂️